# Aplicação Prática de Python na Ciência dos Dados
Aluno: Samuel Marcos

Disciplina: Ciência dos Dados para tomadas de decisões

In [26]:
import pandas as pd

In [27]:
df = pd.read_excel("C:\\Users\\sam-rafa\\Downloads\\default_of_credit_card_clients__courseware_version_1_21_19.xls")

# Analisando os dados 

**Quantas colunas os dados contêm?**

In [28]:
df.shape[1]

25

Através do comando df.shape[1], foi possível verificar que o conjunto de dados possui 25 colunas

**Quantas linhas (amostras)?**

In [29]:
df.shape[0]

30000

Utilizando o comando df.shape[0], foi possível verificar que o conjunto de dados possui 30.000 linhas, onde cada linha representa uma conta de cliente de cartão de crédito.

**Que tipos de características existem. Quais são categóricas e quais são numéricas?**

In [30]:
df.dtypes

ID                            object
LIMIT_BAL                      int64
SEX                            int64
EDUCATION                      int64
MARRIAGE                       int64
AGE                            int64
PAY_1                         object
PAY_2                          int64
PAY_3                          int64
PAY_4                          int64
PAY_5                          int64
PAY_6                          int64
BILL_AMT1                      int64
BILL_AMT2                      int64
BILL_AMT3                      int64
BILL_AMT4                      int64
BILL_AMT5                      int64
BILL_AMT6                      int64
PAY_AMT1                       int64
PAY_AMT2                       int64
PAY_AMT3                       int64
PAY_AMT4                       int64
PAY_AMT5                       int64
PAY_AMT6                       int64
default payment next month     int64
dtype: object

A verificação dos tipos de dados foi realizada utilizando o comando df.dtypes.


O conjunto de dados possui diversas características relacionadas ao perfil do cliente e ao histórico financeiro. Entre elas estão variáveis demográficas como sexo, educação, estado civil e idade; variáveis financeiras como limite de crédito (LIMIT_BAL); histórico de pagamento (PAY_1 até PAY_6); valores das faturas (BILL_AMT1 até BILL_AMT6) e valores pagos (PAY_AMT1 até PAY_AMT6). Além disso, o conjunto de dados possui a variável resposta default payment next month, que indica se o cliente se tornou inadimplente no mês seguinte.

**Características categóricas (classes)**

SEX, EDUCATION, MARRIAGE PAY_1, até PAY_6 e default payment next month.

Elas possuem valores discretos (exemplo: 1, 2, 3 representando categorias).

**Características numéricas**

LIMIT_BAL, AGE, BILL_AMT1 até BILL_AMT6 e PAY_AMT1 até PAY_AMT6.

Essas variáveis representam quantidades numéricas reais.

**Qual é a aparência dos dados segundo essas características?**


In [31]:
df.head()

,ID,LIMIT_BAL,SEX,EDUCATION,MARRIAGE,AGE,PAY_1,PAY_2,PAY_3,PAY_4,...,BILL_AMT4,BILL_AMT5,BILL_AMT6,PAY_AMT1,PAY_AMT2,PAY_AMT3,PAY_AMT4,PAY_AMT5,PAY_AMT6,default payment next month
0,798fc410-45c1,20000,2,2,1,24,2,2,-1,-1,...,0,0,0,0,689,0,0,0,0,1
1,8a8c8f3b-8eb4,120000,2,2,2,26,-1,2,0,0,...,3272,3455,3261,0,1000,1000,1000,0,2000,1
2,85698822-43f5,90000,2,2,2,34,0,0,0,0,...,14331,14948,15549,1518,1500,1000,1000,1000,5000,0
3,0737c11b-be42,50000,2,2,1,37,0,0,0,0,...,28314,28959,29547,2000,2019,1200,1100,1069,1000,0
4,3b7f77cc-dbc0,50000,1,2,1,57,-1,0,-1,0,...,20940,19146,19131,2000,36681,10000,9000,689,679,0


O comando df.head() mostra as 5 primeiras linhas, permitindo observar como os dados estão organizados.

In [32]:
df.describe()

,LIMIT_BAL,SEX,EDUCATION,MARRIAGE,AGE,PAY_2,PAY_3,PAY_4,PAY_5,PAY_6,...,BILL_AMT4,BILL_AMT5,BILL_AMT6,PAY_AMT1,PAY_AMT2,PAY_AMT3,PAY_AMT4,PAY_AMT5,PAY_AMT6,default payment next month
count,30000.000000,30000.000000,30000.000000,30000.000000,30000.000000,30000.000000,30000.000000,30000.000000,30000.000000,30000.000000,...,30000.000000,30000.000000,30000.000000,30000.000000,3.000000e+04,30000.000000,30000.000000,30000.000000,30000.000000,30000.000000
mean,165760.989333,1.586500,1.833333,1.535900,35.108800,-0.132867,-0.164333,-0.219300,-0.263900,-0.288033,...,42791.362167,39884.398167,38480.350933,5613.321500,5.855410e+03,5174.387967,4776.089733,4754.749200,5164.223267,0.218733
std,130158.590432,0.513348,0.807699,0.542698,9.851592,1.191215,1.191096,1.162348,1.127519,1.144981,...,64090.316188,60606.644833,59406.836932,16539.094312,2.299256e+04,17565.538305,15532.893047,15239.070708,17712.664703,0.413394
min,0.000000,0.000000,0.000000,0.000000,0.000000,-2.000000,-2.000000,-2.000000,-2.000000,-2.000000,...,-170000.000000,-81334.000000,-339603.000000,0.000000,0.000000e+00,0.000000,0.000000,0.000000,0.000000,0.000000
25%,50000.000000,1.000000,1.000000,1.000000,28.000000,-1.000000,-1.000000,-1.000000,-1.000000,-1.000000,...,2034.000000,1534.000000,1080.000000,836.000000,7.217500e+02,371.000000,223.000000,170.750000,9.000000,0.000000
50%,140000.000000,2.000000,2.000000,2.000000,34.000000,0.000000,0.000000,0.000000,0.000000,0.000000,...,18759.500000,17835.500000,16643.000000,2084.500000,2.000000e+03,1776.000000,1500.000000,1500.000000,1500.000000,0.000000
75%,240000.000000,2.000000,2.000000,2.000000,41.000000,0.000000,0.000000,0.000000,0.000000,0.000000,...,53572.250000,49804.000000,48863.500000,5000.000000,5.000000e+03,4500.000000,4000.000000,4000.000000,4000.000000,0.000000
max,1000000.000000,2.000000,6.000000,3.000000,79.000000,8.000000,8.000000,8.000000,8.000000,8.000000,...,891586.000000,927171.000000,961664.000000,873552.000000,1.684259e+06,896040.000000,621000.000000,426529.000000,528666.000000,1.000000


o comando df.describe() permite observar o intervalo das variáveis numéricas. Por exemplo, a variável AGE apresenta valores que variam de 0 a 79 anos, com média aproximada de 35 anos.

In [33]:
categoricas = ['SEX','EDUCATION','MARRIAGE','PAY_1','PAY_2','PAY_3','PAY_4','PAY_5','PAY_6','default payment next month']

Criei uma lista com as colunas categoricas.

In [34]:
for col in categoricas:
    print("Coluna:", col)
    print(df[col].value_counts())
    print("--------------------")

Coluna: SEX
SEX
2    17910
1    11775
0      315
Name: count, dtype: int64
--------------------
Coluna: EDUCATION
EDUCATION
2    13884
1    10474
3     4867
0      329
5      275
4      122
6       49
Name: count, dtype: int64
--------------------
Coluna: MARRIAGE
MARRIAGE
2    15810
1    13503
0      369
3      318
Name: count, dtype: int64
--------------------
Coluna: PAY_1
PAY_1
0                13402
-1                5047
1                 3261
Not available     3021
-2                2476
2                 2378
3                  292
4                   63
5                   23
8                   17
6                   11
7                    9
Name: count, dtype: int64
--------------------
Coluna: PAY_2
PAY_2
 0    15878
-1     5989
 2     3878
-2     3746
 3      325
 4       99
 1       28
 5       24
 7       20
 6       12
 8        1
Name: count, dtype: int64
--------------------
Coluna: PAY_3
PAY_3
 0    15914
-1     5869
-2     4046
 2     3782
 3      237
 4       74
 

A frequência das variáveis categóricas foi analisada utilizando o método value_counts() 
aplicado às principais colunas categóricas, como sexo, educação, estado civil e histórico de pagamentos. 
Essa análise permitiu observar a distribuição das categorias.

**Existem dados faltando?**

In [35]:
df.isnull().sum()

ID                            0
LIMIT_BAL                     0
SEX                           0
EDUCATION                     0
MARRIAGE                      0
AGE                           0
PAY_1                         0
PAY_2                         0
PAY_3                         0
PAY_4                         0
PAY_5                         0
PAY_6                         0
BILL_AMT1                     0
BILL_AMT2                     0
BILL_AMT3                     0
BILL_AMT4                     0
BILL_AMT5                     0
BILL_AMT6                     0
PAY_AMT1                      0
PAY_AMT2                      0
PAY_AMT3                      0
PAY_AMT4                      0
PAY_AMT5                      0
PAY_AMT6                      0
default payment next month    0
dtype: int64

foi utilizado o comando df.isnull().sum() para verificar se existe dados faltantes Os resultados indicaram que todas as colunas apresentam valor zero para dados nulos, indicando que não existem dados faltantes.

Foi observado que existem 315 registros onde apenas o campo ID possui valor e as demais variáveis estão zeradas, o que pode ser possíveis inconsistências ou registros incompletos no conjunto de dados.

Durante a análise das variáveis categóricas foi possível identificar a presença de valores que não fazem parte da classificação original do dataset. Por exemplo, na variável SEX foi encontrado o valor 0, embora a codificação oficial utilize apenas 1 (masculino) e 2 (feminino). Situação semelhante ocorre nas variáveis EDUCATION e MARRIAGE, onde aparecem categorias não previstas. Esses resultados indicam possíveis inconsistências ou registros incompletos, sendo necessária uma etapa de limpeza dos dados antes da aplicação de modelos de análise ou predição.